src/gss_names.c

```diff
@@ -281,6 +281,80 @@ uint32_t gssntlm_import_name(uint32_t *minor_status,
                                        output_name);
 }
 
+size_t gssntlm_get_attrs_count(const struct gssntlm_name_attribute *attrs)
+{
+    size_t attrs_count = 0;
+    if (!attrs) return 0;
+    while (attrs[attrs_count].attr_name != NULL) attrs_count++;
+
+    return attrs_count;
+}
+
+int gssntlm_copy_attrs(const struct gssntlm_name_attribute *src,
+                       struct gssntlm_name_attribute **dst)
+{
+    struct gssntlm_name_attribute *copied_attrs = NULL;
+
+    size_t attrs_count = gssntlm_get_attrs_count(src);
```

Comment:
Although this is technically permitted, we try to keep all declarations together, at the start of the function, without empty lines in between.

And have assignments done later.

Reply:
ok